src/modules/metaverse/metaverse.js

```diff
@@ -0,0 +1,119 @@
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+*/
+
+// General Modules
+import Log from '../debugging/log.js';
+
+export class Metaverse {
+    constructor (store, prop) {
+        console.info('constructed', store, prop);
+
+        this.commitLogin = (username, result) => {
+            Log.print('METAVERSE', 'INFO', 'Committing login.');
+            console.info('result', result);
+            const checkIsAdmin = result.account_roles.includes('admin');
+
+            store.commit('mutate', {
+                update: true,
+                property: 'account',
+                with: {
+                    isLoggedIn: true,
+                    isAdmin: checkIsAdmin,
+                    username: username,
+                    accountRoles: result.account_roles,
+                    accountId: result.account_id,
+                    metaverseServer: store.state.metaverseConfig.server,
+                    accessToken: result.access_token,
+                    refreshToken: result.refresh_token,
+                    tokenType: result.token_type,
+                    createdAt: result.created_at,
+                    expiresIn: result.expires_in,
+                    scope: result.scope
+                }
+            });
+        };
+
+        this.logout = () => {
+            store.commit('mutate', {
+                update: true,
+                property: 'account',
+                with: {
+                    username: null,
+                    isLoggedIn: false,
+                    isAdmin: false,
+                    accessToken: null,
+                    refreshToken: null
+                }
+            });
+        };
+    };
+
+    login (metaverse, username, password) {
+        Log.print('METAVERSE', 'INFO', 'Attempting to login as ' + username + '.');
+        return new Promise(function (resolve, reject) {
+            window.$.ajax({
+                type: 'POST',
+                url: metaverse + '/oauth/token',
+                contentType: 'application/x-www-form-urlencoded;charset=UTF-8',
+                data: {
+                    grant_type: 'password',
+                    scope: 'owner', // as opposed to 'domain', we're asking for a user token
+                    username: username,
+                    password: password
+                }
+            })
+                .done(function (data, status, xhr) {
+                    Log.print('METAVERSE', 'INFO', 'Successfully got key and details for ' + username + ' from the Metaverse.');
+                    resolve(data);
+                })
+                .fail(function (xhr, status, errorThrown) {
+                    Log.print('METAVERSE', 'INFO', 'Failed to login as ' + username + ': ' + JSON.stringify(xhr.responseJSON));
+                    reject(xhr.responseJSON);
+                });
+        });
+    };
+
+    commitLogin (username, result) {
```

Comment:
I know this is fixed in the constructor but... if this code actually executed it would always be an infinite loop... drop?